### DISCUSSION TITLE
Palettes & Sprites

### DISCUSSION THREAD

**Discussion by Chonkblonk** [diasurgical/DevilutionX#6888] [Q&A] [answered]
Is there anywhere where I could find Diablo 1's palette files as basically just bitmaps which I could sample directly from in an image editing program? As well as any technical info, like if there's perhaps some sort of limit to how many palettes could be used for any given type of graphic.

I have exactly zero technical knowhow for modifying anything Diablo (it's much more complex than Doom in some senses), but I do some spriting, and I feel like I could try making sprites for some project, and also making variants of existing ones. Altering inventory item graphics would be the easiest as they're single static objects, but there's all kinds of other things one could do.

Wouldn't be right away, as I have some obligations, and I wouldn't perhaps have any direct place to contribute such graphics, nor have the knowledge to implement them (could .PNG's with transparent backgrounds be converted to a format which the .exe wants?), but I could provide them as like a free 'community resource' for any modder to grab and use if they like (be it for a DevilutionX mod or another Diablo mod).

**Comment by DakkJaniels** [comment]
This will probably be helpful for you: https://github.com/diasurgical/d1-graphics-tool
You'll have to unpack your mpq with something (like this: http://www.zezula.net/en/mpq/download.html) to get access to the files.

I'll let someone else more knowledgeable about the graphics speak to the specifics, but that might help get you started with looking into things.

**Comment by AJenbo** [comment]
> if there's perhaps some sort of limit to how many palettes could be used for any given type of graphic.

1

I think this and what Dakk said should get you going, let us know if you have any further questions.

**Comment by Chonkblonk** [comment]
Uh, I do have that program for looking into .MPQs and stuff, because I like to sample from Diablo's sounds for Doom stuff.

The Graphics Tool can export graphics as conventional image formats, but can it **import** and convert those into what Diablo likes?
Presumably, this would work best if the graphic in question adhered to whatever palette restrictions Diablo have, which I could follow much more easily if I could sample directly from those palettes with Photoshop.

EDIT: Oh, you mean that inventory graphics can use only 1 color palette. Are those multiple palettes or is there just a single one for all inventory items?

**Reply by AJenbo** [reply]
> can it import and convert those into what Diablo likes?

yes

> Presumably, this would work best if the graphic in question adhered to whatever palette restrictions Diablo have

It's decent at converting them, I don't bother with that

> which I could follow much more easily if I could sample directly from those palettes with Photoshop.

Does Photoshop not support .pal? Just open it in GIMP and maybe it can export to something Photoshop supports.

**Reply by AJenbo** [reply]
> EDIT: Oh, you mean that inventory graphics can use only 1 color palette. Are those multiple palettes or is there just a single one for all inventory items?

1 palette, the lower 128 colors + the first color (it's what d1gt has as it's palette by default)

**Reply by DakkJaniels** [reply]
Is there support for PNG now? Although I'm not sure how easy it would be to utilize PNG assets without doing code modifications.

**Reply by AJenbo** [reply]
d1gt has png support yes. DevilutionX also has to some extend but only on SDL2 and only for the UI, but at least it's 32bit support so no palette limitation.

**Reply by DakkJaniels** [reply]
Ah ok, yeah I was referring to DevilutionX. I saw some PNG ui assets and wasn't sure.

Regarding D1GT @Chonkblonk , you have to insert/add frames to import images of different formats - just in case you weren't aware - I had some trouble figuring it out how to do it myself.
![image](https://github.com/diasurgical/devilutionX/assets/6080734/b8363af4-9e5c-4cee-aafb-1c60bcc9c171)

**Reply by AJenbo** [reply]
i use drag & drop

**Comment by Chonkblonk** [comment]
Well, what I was vying for was **copies of Diablo's palettes in a conventional image format (.PNG, .BMP, etc),** as well as a method for converting conventional image formats to a format which Diablo can use. That DevilutionX can use .PNGs is great, but I wanted to try to make some assets which _anyone_ modding _any version_ of Diablo could use.

**Reply by DakkJaniels** [reply]
I think there are only 2 palettes (used in dungeon and town), you can easily get them by just saving it from d1gt then convert it to whatever format you want. We already explained the method for converting image formats to CEL using d1gt, not sure what else you need.

**Reply by AJenbo** [reply]
Considering that a palette is not an image I'm not sure I understand why you would want it in a conventional image format?
I think Phothoshop supports .act, .aco and some variant of .pal for opening and saving color palettes.

**Reply by Chonkblonk** [reply]
Mainly because I have a workflow built around using Photoshop and it would be really easy for me to just sample the colors/shades from that directly like it's any other image.
The tool can export the palettes like they're any other image?

EDIT: Sorry if I'm being really obtuse btw.

**Reply by DakkJaniels** [reply, marked as answer]
Export the palette, it will be a .pal file. If Photoshop cannot open that, open the palette in GIMP. Make whatever image you want using the palette colors and save it to whatever image format you want in GIMP, then open in Photoshop.

**Reply by StephenCWills** [reply]
Generally, I would say that both GIMP and Photoshop would open the pal file as a palette, not as an image. Probably, both have a way to visualize the palette, just as d1gt does. So you can just use the `Print Screen` button.

**Reply by AJenbo** [reply]
> Mainly because I have a workflow built around using Photoshop and it would be really easy for me to just sample the colors/shades from that directly like it's any other image. The tool can export the palettes like they're any other image?

Guy walks in to a paint store and asks: "Do you have colors as a painting? I prefer to take my paint from a regular canvas rather then from a tube."

What you are asking doesn't really make a whole lot of sens, would you like the colors in the shape of the monsoonal? A palette is a list of colors, not any form of an image, so there is no regular image to export it as. Here is a video for how to work with palettes in Photoshop in case that is helpful: https://www.youtube.com/watch?v=IK4Ftiqt8pE
